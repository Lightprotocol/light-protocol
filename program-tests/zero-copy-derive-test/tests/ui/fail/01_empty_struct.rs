// Edge case: Empty struct (unit struct)

use light_zero_copy_derive::{ZeroCopy, ZeroCopyMut};

#[derive(ZeroCopy, ZeroCopyMut)]
#[repr(C)]
pub struct EmptyStruct;

fn main() {}
