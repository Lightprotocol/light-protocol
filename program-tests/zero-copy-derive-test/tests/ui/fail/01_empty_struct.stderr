error: ZeroCopy only supports structs with named fields
 --> tests/ui/fail/01_empty_struct.rs:5:10
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `ZeroCopy` (in Nightly builds, run with -Z macro-backtrace for more info)

error: ZeroCopy only supports structs with named fields
 --> tests/ui/fail/01_empty_struct.rs:5:20
  |
5 | #[derive(ZeroCopy, ZeroCopyMut)]
  |                    ^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)
